google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/ManagedCloudSdkService.java

```diff
@@ -175,27 +176,33 @@ private boolean executeManagedSdkJob(ManagedSdkJobType jobType, Callable<Path> m
    * @throws InterruptedException if Managed Cloud SDK has been interrupted by {@link
    *     #cancelInstallOrUpdate()} ()}
    */
-  private Path installManagedSdk() throws Exception {
-    Path installedManagedSdkPath = installSdk();
-    installAppEngineJavaComponent();
-
-    return installedManagedSdkPath;
+  private ManagedSdkJobResult installManagedSdk() throws Exception {
+    ManagedSdkJobResult installResult = installSdk();
+    ManagedSdkJobResult appEngineJavaResult = installAppEngineJavaComponent();
+
+    // return up-to-date only if both SDK and app engine Java were up-to-date
+    return (installResult == ManagedSdkJobResult.UP_TO_DATE
```

Comment:
maybe we don't solve this now, but if the SDK was already installed, but the java component was not, the user will still see the same message saying the SDK was installed right?

Reply:
correct, you mean we might show a slightly different message in this case?

Reply:
I don't think its a huge deal, but it may be confusing to some users if they already had the SDK installed and they see a message saying it was installed successfully (even though behind the scenes it just installed the java component). 

I'll approve the PR.

Reply:
I think the inconvenience of this is slightly diminished by the fact that this only applies to SDK we manage ourselves, so the chance of install being completed but Java component missing must be really rare.

Reply:
ah good point. didn't consider that. we always install them both right? that makes sense then.